Model/AdyenOrderPaymentStatus.php

```diff
@@ -75,12 +75,17 @@ public function __construct(
 
     /**
      * @param string $orderId
+     * @param string $shopperEmail
      * @return bool|string
      */
-    public function getOrderPaymentStatus($orderId)
+    public function getOrderPaymentStatus($orderId, $shopperEmail)
     {
         try {
             $order = $this->orderRepository->get($orderId);
+            // Validate shopper email against order email to prevent abuse
```

Comment:
Maybe update the comment to: `Validate shopper email against order email for extra security`?

Reply:
Makes sense
```suggestion
            // Validate shopper email against order email for extra security.
```